Source/WebCore/page/Navigation.cpp

```diff
@@ -424,7 +425,7 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
         downloadRequestFilename,
         info,
         canIntercept,
-        false, // FIXME: userInitiated
+        UserGestureIndicator::processingUserGesture(),
```

Comment:
Do click actions count as user gestures? Like on `a` elements?

Reply:
Yes AFAIK (unlike a.click() which creates a simulated mouse event).

Reply:
Don't we need to pass the Document pointer in here? Otherwise we will never get a `true` result I think.

Reply:
Oh whoops that's not right. We can not pass in the document and still get true.

Reply:
In fact it looks like passing the default document as null can/will return true due to the last statement in the function. But it does not seem the right thing indeed, I will try passing the document and check that the test still passes.

Reply:
@rreno we came to the same conclusion, but it still looks like passing the document is the way to go.